docs/dev_contribution.rst

```diff
@@ -0,0 +1,95 @@
+.. Contribution documentation for libCellML
+
+============
+Contributing
+============
+
+This document covers the process to follow for getting your changes into the :term:`prime repository`.  While there are many types of contribution, this section focuses on contributions made through GitHub and Git, or in other words assets that are managed using the version control system.  It is assumed that :doc:`Setup <dev_setup>` and :doc:`Building <dev_building>` have already been read and followed. 
+
+.. contents::
+
+Overview
+========
+
+For any body of work intended for the :term:`prime repository` start with a GitHub issue.  The issue can be used to discuss the topic and clarify any problems related to it.  Once progress has been made towards addressing the issue a pull request is created that references the issue.  
+
+Reviewers provide feedback on the changes by adding comments to the pull request or associated commits. The BuildBot build/test procedure will run each time changes are pushed to the pull request's branch, and the results are displayed in the pull request view.
+
+Once all the changes and reviews are complete, one of the :term:`prime repository` owners will merge the pull request into the prime repository, onto the develop branch.
+
+Note that a bug is just a type of issue, and that resolving the bug should have both the implementation to fix the bug and a test that triggers the bug.
+
+:numref:`Figure %s <libcellml-development-process>` gives a graphical overview of the developer contribution process.  For more detail see the text below.
+
+.. _libcellml-development-process:
+.. figure:: images/libCellMLProcesses-DevelopmentProcess.png
+   :align: center
+   :alt: Developer contribution process
+
+   Developer contribution process
+
+GitHub Issue
+============
+
+If an issue does not exist for the required work (e.g. implementation of a feature, fixing of a bug), then create a new one.  The issue is the place to discuss the particulars related to the issue, discussions on determining the scope of the issue or clarification of any points that are unclear.
+
+Topic Branch
+============
+
+A topic or feature branch is a branch that is local to you (and anyone you collaborate with), it is branch that will not be available from the :term:`prime repository`.  All development work should be carried out on a topic branch, for example any major feature that you work on or minor bug fix.  Before creating a local topic branch pull the latest changes from the :term:`prime repository` develop branch.
+
+.. _libcellml-branching:
+.. figure:: images/libCellMLProcesses-GitBranching.png
```

Comment:
Shouldn't the **pull/fetch** dashed arrow between the **CellML** and **PC** 'boxes' be going from **CellML** to **PC** and not the other way round?